src/Symfony/Component/Clock/MockClock.php

```diff
@@ -46,15 +46,15 @@ public function sleep(float|int $seconds): void
             $now[0] += $s;
         }
 
-        if (0 < ($us = $seconds - $s) && 1E6 <= $now[1] += $us * 1E6) {
+        if (0 < ($us = $seconds - $s) && 1E6 <= $now[1] += round($us * 1E6)) {
             ++$now[0];
             $now[1] -= 1E6;
         }
 
         $datetime = '@'.$now[0].'.'.str_pad($now[1], 6, '0', \STR_PAD_LEFT);
         $timezone = $this->now->getTimezone();
 
-        $this->now = (new \DateTimeImmutable($datetime, $timezone))->setTimezone($timezone);
+        $this->now = (new \DateTimeImmutable($datetime))->setTimezone($timezone);
```

Comment:
Please keep this line, passing a second argument makes the calla bit faster

Reply:
You are storehouse of knowledge! Did you know?

Reply:
;)